src/index.ts

```diff
@@ -116,6 +118,16 @@ export async function run() {
     let sourceAccountId: string;
     let webIdentityToken: string;
 
+    //if the user wants to attempt to use existing credentials, check if we have some already
+    if (useExistingCredentials === 'true') {
```

Comment:
You're using `===` to compare to the string value of `'true'` but I'm pretty sure that's not what you want.